Invalid cassandra-reaper.yml generated when REAPER_JMX_AUTH_PASSWORD contains special chars

Reaper version was 2.3.0.

The script `configure-jmx-credentials.sh` does not enclose some env vars in double quotes when printing them to /etc/cassandra-reaper.yml, see [here](https://github.com/thelastpickle/cassandra-reaper/blob/b9c7129361db47da70b56ca589db36929939e891/src/server/src/main/docker/configure-jmx-credentials.sh#L46-L47).

This is a problem, especially when `REAPER_JMX_AUTH_PASSWORD` contains invalid chars:

```
io.dropwizard.configuration.ConfigurationParsingException: /etc/cassandra-reaper.yml has an error:
  * Malformed YAML at line: 99, column: 14; while scanning for the next token; found character %'%' that cannot start any token;  in 'reader', line 98, column 15:
        password: %MpG'+A%n3j9|oG4dXNc
                  ^
 at [Source: (ByteArrayInputStream); line: 98, column: 13]

	at io.dropwizard.configuration.ConfigurationParsingException$Builder.build(ConfigurationParsingException.java:279)
	at io.dropwizard.configuration.BaseConfigurationFactory.build(BaseConfigurationFactory.java:96)
	at io.dropwizard.cli.ConfiguredCommand.parseConfiguration(ConfiguredCommand.java:126)
	at io.dropwizard.cli.ConfiguredCommand.run(ConfiguredCommand.java:74)
	at io.dropwizard.cli.Cli.run(Cli.java:78)
	at io.dropwizard.Application.run(Application.java:93)
	at io.cassandrareaper.ReaperApplication.main(ReaperApplication.java:109)
Caused by: com.fasterxml.jackson.dataformat.yaml.snakeyaml.error.MarkedYAMLException: while scanning for the next token; found character %'%' that cannot start any token;  in 'reader', line 98, column 15:
        password: %MpG'+A%n3j9|oG4dXNc
```

--- a/src/server/src/main/docker/configure-jmx-credentials.sh
+++ b/src/server/src/main/docker/configure-jmx-credentials.sh
@@ -14,7 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# we expect the jmx credentials to be a comma-separated list of 'user:passowrd@cluster' entries
+# we expect the jmx credentials to be a comma-separated list of 'user:password@cluster' entries
 if [ ! -z "${REAPER_JMX_CREDENTIALS}" ]; then
 
 cat <<EOT >> /etc/reaper/cassandra-reaper.yml
@@ -25,14 +25,14 @@ EOT
   for ENTRY in $(echo "${REAPER_JMX_CREDENTIALS}" | sed "s/,/ /g"); do
     # and then just cut out the fields we need
     CLUSTER=$(echo "${ENTRY}" | cut -d'@' -f2)
-    USERNAME=$(echo "${ENTRY}" | cut -d'@' -f1 | cut -d':' -f1)
-    PASSWORD=$(echo "${ENTRY}" | cut -d'@' -f1 | cut -d':' -f2)
+    USERNAME=$(echo "${ENTRY}" | cut -d'@' -f1 | cut -d':' -f1 | sed 's/"/\\"/g')
+    PASSWORD=$(echo "${ENTRY}" | cut -d'@' -f1 | cut -d':' -f2 | sed 's/"/\\"/g')
 
     # finally, write out the YAML entries
 cat <<EOT >> /etc/reaper/cassandra-reaper.yml
   ${CLUSTER}:
-    username: ${USERNAME}
-    password: ${PASSWORD}
+    username: "${USERNAME}"
+    password: "${PASSWORD}"
 EOT
 
   done
@@ -43,8 +43,8 @@ fi
 if [ ! -z "${REAPER_JMX_AUTH_USERNAME}" ]; then
 cat <<EOT >> /etc/reaper/cassandra-reaper.yml
 jmxAuth:
-  username: ${REAPER_JMX_AUTH_USERNAME}
-  password: ${REAPER_JMX_AUTH_PASSWORD}
+  username: "$(echo "${REAPER_JMX_AUTH_USERNAME}" | sed 's/"/\\"/g')"
+  password: "$(echo "${REAPER_JMX_AUTH_PASSWORD}" | sed 's/"/\\"/g')"
 EOT
 
 fi

--- a/src/server/src/main/docker/configure-persistence.sh
+++ b/src/server/src/main/docker/configure-persistence.sh
@@ -62,8 +62,8 @@ if [ "true" = "${REAPER_CASS_AUTH_ENABLED}" ]; then
 cat <<EOT >> /etc/reaper/cassandra-reaper.yml
   authProvider:
     type: plainText
-    username: ${REAPER_CASS_AUTH_USERNAME}
-    password: ${REAPER_CASS_AUTH_PASSWORD}
+    username: "$(echo "${REAPER_CASS_AUTH_USERNAME}" | sed 's/"/\\"/g')"
+    password: "$(echo "${REAPER_CASS_AUTH_PASSWORD}" | sed 's/"/\\"/g')"
 EOT
 fi
 
@@ -99,4 +99,3 @@ fi
 
     ;;
 esac
-

